File: Nusa Carbon Mobile/NusaCarbon 2/src/main/java/com/nusacarbon/api/entity/Certificate.java
package com.nusacarbon.api.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "certificates")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Certificate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idSertifikat;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_retirement", unique = true, nullable = false)
    private Retirement retirement;

    @Column(nullable = false, unique = true, length = 100)
    private String nomorSertifikat;

    @Column(nullable = false, length = 200)
    private String namaEntitas;

    @Column(nullable = false, precision = 14, scale = 4)
    private BigDecimal totalCo2e;

    @Column(length = 500)
    private String linkFilePdf;

    @Column(length = 255)
    private String nftTokenId;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;
}

File: Nusa Carbon Mobile/nusacarbon_app/nusacarbon/lib/models/certificate.dart
class Certificate {
  final int idSertifikat;
  final int idRetirement;
  final String nomorSertifikat; // "CERT-NC-2024-001"
  final String namaEntitas;
  final double totalCo2e;
  final String? linkFilePdf;
  final String? nftTokenId;
  final DateTime createdAt;

  const Certificate({
    required this.idSertifikat,
    required this.idRetirement,
    required this.nomorSertifikat,
    required this.namaEntitas,
    required this.totalCo2e,
    this.linkFilePdf,
    this.nftTokenId,
    required this.createdAt,
  });

  factory Certificate.fromJson(Map<String, dynamic> json) {
    return Certificate(
      idSertifikat: json['id_sertifikat'] as int,
      idRetirement: json['id_retirement'] as int,
      nomorSertifikat: json['nomor_sertifikat'] as String,
      namaEntitas: json['nama_entitas'] as String,
      totalCo2e: (json['total_co2e'] as num).toDouble(),
      linkFilePdf: json['link_file_pdf'] as String?,
      nftTokenId: json['nft_token_id'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_sertifikat': idSertifikat,
      'id_retirement': idRetirement,
      'nomor_sertifikat': nomorSertifikat,
      'nama_entitas': namaEntitas,
      'total_co2e': totalCo2e,
      'link_file_pdf': linkFilePdf,
      'nft_token_id': nftTokenId,
      'created_at': createdAt.toIso8601String(),
    };
  }
}
